File: JavaScript/Linked Lists/LinkedListCycle.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */

/**
 * @param {ListNode} head
 * @return {boolean}
 */
var hasCycle = function(head) {
    let fast = head; 
    while(fast && fast.next){
        fast = fast.next.next; 
        head = head.next;
        if(fast === head){ //If the two nodes overlap, then the list is circular
            return true;
        }
    }
    return false; 
};

File: Java/Linked Lists/LinkedListCycle.java
import java.util.HashSet;

/**
 * Definition for singly-linked list.
**/ class ListNode {
      int val;
      ListNode next;
      ListNode(int x) {
          val = x;
          next = null;
      }
  }

public class LinkedListCycle {
    public boolean hasCycle(ListNode head) {
        HashSet <ListNode> visited = new HashSet<>(); 
        while (head != null){
            if(visited.contains(head)){
                return true; 
            }
            visited.add(head); 
            head = head.next;
        }
        return false; 
    }
}